### PR TITLE
Various improvements.

### PR DESCRIPTION
**Pull request by asiekierka**
* Verified KTR-001 and JAN-001 console units.
* Added on-device SHA-256 calculation. The implementation is dedicated to the public domain (7zip LZMA SDK) and manages to compute the checksum in about one second.
* Moved printf() calls to iprintf() calls - the latter don't include floating-point support code and are, as such, somewhat smaller; allowing bios-dump.gba to fit within 128K even with SHA-256 calculation logic added.

### CODE REVIEW

File: src/bios-dump.c [hunk 1]
```diff
@@ -1,9 +1,15 @@
+#define BIOS_WRITE_SRAM
+#define BIOS_CALC_SHA256
+
```
Comment (endrift) [inline src/bios-dump.c]: This should be in the makefile

File: src/Sha256.c [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+This code is based on public domain code from Wei Dai's Crypto++ library. */
+
+#include <gba_types.h>
+#include <string.h>
+
+#include "Sha256.h"
+
+// #include "Precomp.h"
+// #include "CpuArch.h"
+// #include "RotateDefs.h"
```
Comment (endrift) [inline src/Sha256.c]: These should probably be removed

File: src/Sha256.c [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+  state[6] += g;
+  state[7] += h;
+  #else
+  for (j = 0; j < 8; j++)
+    state[j] += T[j];
+  #endif
+  
+  /* Wipe variables */
+  /* memset(W, 0, sizeof(W)); */
+  /* memset(T, 0, sizeof(T)); */
```
Comment (endrift) [inline src/Sha256.c]: Commented out code should be removed.

**Review by endrift — COMMENTED** [review]
It might make sense to add a list of known SHA-256 sums to compare against too.

File: src/Sha256.c (hunk 3 above)
Comment (asiekierka) [reply src/Sha256.c]: Ah, I would have missed that one - it was actually commented out upstream.

**Comment by asiekierka** [comment]
Should be taken care of, albeit with only two checksums I decided not to make it a list for now.

**Pull request merged**